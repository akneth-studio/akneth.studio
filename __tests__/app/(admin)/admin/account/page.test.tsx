/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import Account from '@/app/(admin)/admin/account/page';

// Mock Supabase client
jest.mock('@/utils/supabase/client', () => ({
  supabase: {
    auth: {
      getUser: jest.fn(),
      updateUser: jest.fn(),
    },
  },
}));

describe('Account Page', () => {
  const mockGetUser = jest.fn();
  const mockUpdateUser = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    
    // Get the actual mock functions from the module
    const { supabase } = require('@/utils/supabase/client');
    Object.assign(mockGetUser, supabase.auth.getUser);
    Object.assign(mockUpdateUser, supabase.auth.updateUser);
    
    // Default mock implementations
    mockGetUser.mockResolvedValue({
      data: {
        user: {
          id: 'user123',
          email: 'test@example.com',
          user_metadata: {
            display_name: 'Test User'
          }
        }
      }
    });
    
    mockUpdateUser.mockResolvedValue({ error: null });
  });

  it('renders account information', async () => {
    render(<Account />);
    
    await waitFor(() => {
      expect(screen.getByText('Moje konto')).toBeInTheDocument();
      expect(screen.getByText('user123')).toBeInTheDocument();
      expect(screen.getByText('test@example.com')).toBeInTheDocument();
      expect(screen.getByText('Test User')).toBeInTheDocument();
    });
  });

  it('renders correctly', async () => {
    const { container } = render(<Account />);
    
    await waitFor(() => {
      expect(screen.getByText('Moje konto')).toBeInTheDocument();
    });
    
    expect(container).toMatchSnapshot();
  });

  it('updates display name', async () => {
    render(<Account />);
    
    // Wait for initial data to load
    await waitFor(() => {
      expect(screen.getByText('Test User')).toBeInTheDocument();
    });
    
    // Fill in new display name
    const displayNameInput = screen.getByPlaceholderText('Imię i nazwisko lub pseudonim');
    fireEvent.change(displayNameInput, { target: { value: 'New Name' } });
    
    // Submit form
    const saveButton = screen.getByText('Zapisz nazwę konta');
    fireEvent.click(saveButton);
    
    await waitFor(() => {
      expect(mockUpdateUser).toHaveBeenCalledWith({ 
        data: { display_name: 'New Name' } 
      });
    });
  });

  it('changes email', async () => {
    render(<Account />);
    
    // Fill in new email
    const emailInput = screen.getByPlaceholderText('Podaj nowy e-mail');
    fireEvent.change(emailInput, { target: { value: 'new@example.com' } });
    
    // Submit form
    const changeEmailButton = screen.getByText('Zmień e-mail');
    fireEvent.click(changeEmailButton);
    
    await waitFor(() => {
      expect(mockUpdateUser).toHaveBeenCalledWith({ 
        email: 'new@example.com' 
      });
    });
  });

  it('changes password', async () => {
    render(<Account />);
    
    // Fill in password fields
    const passwordInput = screen.getByPlaceholderText('Nowe hasło');
    const confirmPasswordInput = screen.getByPlaceholderText('Powtórz nowe hasło');
    
    fireEvent.change(passwordInput, { target: { value: 'test-password' } });
    fireEvent.change(confirmPasswordInput, { target: { value: 'test-password' } });
    
    // Submit form
    const changePasswordButton = screen.getByText('Zmień hasło');
    fireEvent.click(changePasswordButton);
    
    await waitFor(() => {
      expect(mockUpdateUser).toHaveBeenCalledWith({ 
        // deepcode ignore NoHardcodedPasswords/test: test file
        password: 'test-password' 
      });
    });
  });

  it('shows error when passwords do not match', async () => {
    render(<Account />);
    
    // Fill in password fields with different values
    const passwordInput = screen.getByPlaceholderText('Nowe hasło');
    const confirmPasswordInput = screen.getByPlaceholderText('Powtórz nowe hasło');
    
    fireEvent.change(passwordInput, { target: { value: 'test-password' } });
    fireEvent.change(confirmPasswordInput, { target: { value: 'different-test-password' } });
    
    // Submit form
    const changePasswordButton = screen.getByText('Zmień hasło');
    fireEvent.click(changePasswordButton);
    
    await waitFor(() => {
      expect(screen.getByText('Hasła muszą być identyczne.')).toBeInTheDocument();
    });
    
    expect(mockUpdateUser).not.toHaveBeenCalled();
  });

  it('shows error when display name is empty', async () => {
    render(<Account />);
    
    // Wait for initial data to load
    await waitFor(() => {
      expect(screen.getByText('Test User')).toBeInTheDocument();
    });
    
    // Clear the display name input
    const displayNameInput = screen.getByPlaceholderText('Imię i nazwisko lub pseudonim');
    fireEvent.change(displayNameInput, { target: { value: '' } });
    
    // Get the form and submit it
    const form = screen.getByText('Twoja nazwa konta').closest('form');
    if (form) {
      fireEvent.submit(form);
    }
    
    // Wait for error message to appear
    await waitFor(() => {
      expect(screen.getByText('Podaj nazwę konta.')).toBeInTheDocument();
    }, { timeout: 2000 });
    
    expect(mockUpdateUser).not.toHaveBeenCalled();
  });
});
