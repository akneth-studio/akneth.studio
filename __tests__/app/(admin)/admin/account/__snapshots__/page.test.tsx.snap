// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Account Page renders correctly 1`] = `
<div>
  <div
    class="py-5 container"
    style="max-width: 480px;"
  >
    <h2
      class="mb-4 text-center"
    >
      Moje konto
    </h2>
    <div
      class="mb-4"
    >
      <p>
        <strong>
          ID:
        </strong>
         
      </p>
      <p>
        <strong>
          E-mail:
        </strong>
         
      </p>
      <p>
        <strong>
          Nazwa konta:
        </strong>
         
      </p>
    </div>
    <form
      class="mb-4"
    >
      <label
        class="form-label"
      >
        Twoja nazwa konta
      </label>
      <input
        class="form-control"
        placeholder="Imię i nazwisko lub pseudonim"
        required=""
        type="text"
        value=""
      />
      <button
        class="mt-3 w-100 btn btn-primary"
        type="submit"
      >
        Zapisz nazwę konta
      </button>
    </form>
    <form
      class="mb-4"
    >
      <label
        class="form-label"
      >
        Nowy e-mail
      </label>
      <input
        class="form-control"
        placeholder="Podaj nowy e-mail"
        type="email"
        value=""
      />
      <button
        class="mt-2 w-100 btn btn-primary"
        type="submit"
      >
        Zmień e-mail
      </button>
    </form>
    <form
      class="mb-4"
    >
      <label
        class="form-label"
      >
        Nowe hasło
      </label>
      <input
        class="form-control"
        minlength="8"
        placeholder="Nowe hasło"
        type="password"
        value=""
      />
      <label
        class="mt-2 form-label"
      >
        Powtórz hasło
      </label>
      <input
        class="form-control"
        minlength="8"
        placeholder="Powtórz nowe hasło"
        type="password"
        value=""
      />
      <button
        class="mt-2 w-100 btn btn-primary"
        type="submit"
      >
        Zmień hasło
      </button>
    </form>
  </div>
</div>
`;
